service/snap/snap.go

```diff
@@ -39,6 +39,13 @@ var (
 	snapNameRe = regexp.MustCompile("^[a-z0-9][a-z0-9-]{0,39}[^-]$")
 )
 
+// Runnable expects to be able to run a given command with a series of arguments
+// and return the output and/or error from that executing command.
+type Runnable interface {
+	Execute(name string, args ...string) (string, error)
+	Clock() clock.Clock
```

Comment:
Why would a `Runnable` thing give you access to a `clock.Clock`?  The `Clock` method seems out of place in this interface.

Reply:
Agreed, was being cheeky here.